### PR TITLE
Pointer mode clean and fix.

### PR DESCRIPTION
**Pull request by taiyu-len**
cleaned up and fixed all the mouse resize/drag code.
no longer any odd behavior, many of which are caused by changing what view is being resized (specially with floating windows),
which sometimes involved odd behavior of resizing the workspace, or any random container.

i couldnt fit the resizing code into resize.c without making a function take a ton of variables,
but on the plus side i reduced `mouse_resize_tiled` from 170 loc to 32 loc. (with less odd behavior too)
floating resize is about the same too, but lacks the bugs (like resizing the workspace)

tl;dr
everything we need is set when a mode starts, (on button press),
and we update with that information (on button motion)

### CODE REVIEW

File: sway/input_state.c [hunk 1]
```diff
@@ -48,27 +48,261 @@ void release_key(keycode key) {
...
+
+void pointer_mode_reset(void) {
+	switch (pointer_state.mode) {
+	case M_FLOATING | M_DRAGGING:
+	case M_FLOATING | M_RESIZING:
+		reset_initial_view();
+		break;
+
+	case M_TILING | M_DRAGGING:
+	case M_TILING | M_RESIZING:
```
Comment (taiyu-len) [inline sway/input_state.c]: might reset tiling resize the same way as floating, all the information we need is there. just like 5 loc

File: sway/layout.c [hunk 2]
```diff
@@ -154,6 +203,30 @@ void move_container(swayc_t *container,swayc_t* root,enum movement_direction dir
...
+	struct wlc_geometry geometry = {
+		.origin = {
+			.x = container->x + container->gaps / 2,
+			.y = container->y + container->gaps / 2
+		},
+		.size = {
+			.w = container->width - container->gaps,
+			.h = container->height - container->gaps
+		}
+	};
```
Comment (taiyu-len) [inline sway/layout.c]: forgot to add conditional `if is_floating` so it doesnt add gaps when it doesnt need to,
also forgot to reuse this in arrange_windows for the floating windows.

**Comment by Luminarys** [comment]
I'm not sure how extensively you've tested the code for the tiling resizes, but they're definitely not functioning as they should compared to prior(try wildly flinging your mouse around and you'll see weird behavior). However, given the complexity of the previous resizing code, I think the best way to handle minimum width/height would be to go with these changes and try to get wlc to expose a function to allow us to set the pointer position, so if min sane height/width were exceeded the pointer's dy/dx could be reverted.

**Comment by taiyu-len** [comment]
ah ya,
there is some oddities with resize. ill try to get that fixed.

**Comment by Luminarys** [comment]
Tested again, LGTM. :+1:

File: include/input_state.h [hunk 3]
```diff
@@ -27,34 +28,55 @@ enum pointer_values {
 	M_SCROLL_DOWN = 276,
 };
 
+enum pointer_mode {
+	// Target
+	M_FLOATING = 1 << 0,
+	M_TILING = 1 << 1,
+	// Action
+	M_DRAGGING = 1 << 2,
+	M_RESIZING = 1 << 3,
```
Comment (SirCmpwn) [inline include/input_state.h]: Style: I prefer to declare these with powers of two explicitly, like so:

    M_FLOATING = 1,
    M_TILING = 2,
    M_DRAGGING = 4,
    M_RESIZING = 8

File: sway/handlers.c [hunk 4]
```diff
@@ -334,114 +331,109 @@ static bool handle_key(wlc_handle view, uint32_t time, const struct wlc_modifier
...
-	swayc_t *focused = get_focused_container(&root_container);
-	// dont change focus if fullscreen
-	if (swayc_is_fullscreen(focused)) {
-		return false;
-	}
-	if (state == WLC_BUTTON_STATE_PRESSED) {
-		sway_log(L_DEBUG, "Mouse button %u pressed", button);
-		if (button == M_LEFT_CLICK) {
-			pointer_state.l_held = true;
+	enum { DONT_SEND_CLICK = true, SEND_CLICK = false };
```
Comment (SirCmpwn) [inline sway/handlers.c]: What is with you and using enums instead of defines? Justify this behaviour or switch to defines.

File: sway/input_state.c [hunk 5]
```diff
@@ -48,27 +48,267 @@ void release_key(keycode key) {
...
+		} sib;
+	} lr;
+	struct {
+		double y, h;
+		swayc_t *ptr;
+		struct {
+			double y, h;
+			swayc_t *ptr;
+		} sib;
+	} tb;
```
Comment (SirCmpwn) [inline sway/input_state.c]: These names "lr", "tb", "sib", are not very helpful.

**Comment by SirCmpwn** [comment]
Fix the issues I've brought up and let me know when it's ready. I'll test this then.

File: sway/handlers.c (hunk 4 above)
Comment (taiyu-len) [inline sway/handlers.c]: no need to undef after, it stays within the scope.
not really any other reason.

Comment (SirCmpwn) [inline sway/handlers.c]: I don't like it. How about we have broader defines for `EVENT_PASSTHROUGH` and `EVENT_HANDLED` or something

**Comment by taiyu-len** [comment]
@SirCmpwn i should probably post that its done i guess.

**Comment by SirCmpwn** [comment]
Okay. I'll test this on top of X11. Can anyone test it on DRM?

**Comment by SirCmpwn** [comment]
Output 1 resolution changed to 5760 x 1080
    Added output 1:WLC-0
    sway: /tmp/yaourt-tmp-sircmpwn/aur-wlc-git/src/wlc/src/compositor/output.c:636: wlc_output_set_resolution_ptr: Assertion `resolution->w != 0 && resolution->h != 0' failed.

    Program received signal SIGABRT, Aborted.
    0x00007ffff73e45f8 in raise () from /usr/lib/libc.so.6

**Comment by taiyu-len** [comment]
well that was a bug from the point i branched from, merged so it should be fine.

**Comment by SirCmpwn** [comment]
LGTM
